paymentsheet/src/main/java/com/stripe/android/paymentsheet/repositories/StripeCustomerAdapter.kt

```diff
@@ -26,6 +26,9 @@ internal class StripeCustomerAdapter @Inject constructor(
     private val prefsRepositoryFactory: (CustomerEphemeralKey) -> PrefsRepository,
 ) : CustomerAdapter {
 
+    internal var cachedCustomer: Result<CustomerEphemeralKey>? = null
```

Comment:
What's the threading story here (long term). Do we expect to set this from different threads? Read from different threads? Everything from main thread?

Reply:
This should be only called from the suspending functions in the adapter.

Reply:
Sounds like we need to update this to be more concurrent friendly then.

I'd be happy to tackle this as a follow up if you'd like.